### PR TITLE
Tag changes for cuda-quantum image and signing

### PR DESCRIPTION
**Pull request by bettinaheim**
The tags for the cuda-quantum image change as follows:
- the tag `latest-hpc` is replaced by the `latest` tag
- the tag `latest` is replaced by the `latest-base` tag

The push to NGC is still running into permissions issues that need to be resolved.
I set up the pipelines so the push to NGC is simply skipped when NGC_CREDENTIALS are not defined as a secret in the ghcr-deployment environment (will rename that environment to simply deployment in the near future). I removed the secret from the repo for now.

Once the correct secret is stored on the repo, the pipelines should start pushing the cuda-quantum `latest` and `latest-base` images to NGC, as well as the version from (future) release branches. 

Container signing using cosign:
To verify an image e.g. use a Dockerfile
```
# Source: https://github.com/sigstore/cosign
FROM gcr.io/projectsigstore/cosign as cosign-bin

# Source: https://github.com/chainguard-images/
FROM cgr.dev/chainguard/static:latest
COPY --from=cosign-bin /ko-app/cosign /usr/local/bin/cosign
ENTRYPOINT [ "/usr/local/bin/cosign" ]
```
Then execute the commands:

```
docker build -t cosign .
image_name=ghcr.io/nvidia/cuda-quantum-devdeps
image_tag=ext-amd64-gcc12-workflows-ci
docker run --rm cosign verify $image_name:$image_tag --certificate-oidc-issuer https://token.actions.githubusercontent.com --certificate-identity-regexp 'https://github.com/NVIDIA/cuda-quantum/.github/workflows/\S+\.yml@\S+' | jq .
```

### CODE REVIEW

File: .github/workflows/publishing.yml [hunk 1]
```diff
@@ -200,16 +200,19 @@ jobs:
...
           docker image rm $cudaq_image
           docker image prune --force
 
           echo "release_id=$release_id" >> $GITHUB_OUTPUT
-          echo "image_name=$image_name" >> $GITHUB_OUTPUT
-          echo "image_tag=${base_tag}-hpc" >> $GITHUB_OUTPUT
+          echo "ngc_registry=$ngc_registry" >> $GITHUB_OUTPUT
+          echo "image_short_name=$image_short_name" >> $GITHUB_OUTPUT
+          echo "image_name=${ngc_registry:-$registry}/$image_short_name" >> $GITHUB_OUTPUT
+          echo "image_tag=${base_tag#-base}" >> $GITHUB_OUTPUT
```
Comment (bettinaheim) [inline .github/workflows/publishing.yml:215]:
```suggestion
          echo "image_tag=${base_tag%-base}" >> $GITHUB_OUTPUT
```

File: .github/workflows/deployments.yml [hunk 2]
```diff
@@ -119,8 +119,12 @@ jobs:
           environment=${{ (github.event_name != 'workflow_run' && 'ghcr-deployment') || '' }}
 
           # Store deployment info
+          release_title="${{ inputs.create_release }}"
+          release_version=`echo "$release_title" | (egrep -o "([0-9]{1,}\.)+[0-9]{1,}([A-Za-z0-9_\-\.]*)" || true)`
+
           echo "source-sha: ${{ github.sha }}" >> deployment_info.txt
-          echo "publish: ${{ inputs.create_release }}" >> deployment_info.txt
+          echo "release-title: $release_title" >> deployment_info.txt
```
[the remark this answers is not in the document]
Comment (bettinaheim) [reply .github/workflows/deployments.yml:126]: I am not sure I quite understand the question. The release title is set to the argument provided when manually invoking the workflow. This argument is optional and when supplied triggers the creation of an actual release, like e.g. 0.4.1.-alpha. I don't think there is a convenient way to enforce a particular format, but the documentation of the argument says what is expected. The distinction of the version and the title is maybe unnecessary, but I also don't see an issue with being a bit more permissive in the title and allowing to call a release e.g. v0.4.1 when the tag/version is 0.4.1.

**Comment by bettinaheim** [comment]
Manual deployment (without NGC push): https://github.com/NVIDIA/cuda-quantum/actions/runs/5906892320
And the corresponding publishing: https://github.com/NVIDIA/cuda-quantum/actions/runs/5907474908/

File: .github/workflows/deployments.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmhowe23) [reply .github/workflows/deployments.yml:126]: I was just double checking that it was indeed intentional to have two nearly identical strings being saved that way. I will mark it as resolved.

**Comment by bettinaheim** [comment]
I'll pull this in so NGC pushes can be enabled while I am out. More details shared with the repo maintainers.

**Pull request merged**